packages/mask/src/plugins/FindTruman/SNSAdaptor/EncryptionCard.tsx

```diff
@@ -1,68 +1,95 @@
+            '& h1, & h2, & h3, & h4, & h5, & h6': {
+                fontSize: 14,
+            },
+            '& img': {
+                maxWidth: '100%',
+            },
+            '& a': {
+                color: theme.palette.text.primary,
+            },
+        },
+    }
+})
+
+const getEncryptionError = (t: FindTrumanI18nFunction, errorCode: number) => {
+    switch (errorCode) {
+        case 102:
+            return t('plugin_find_truman_decrypt_102')
+        case 1004:
+            return t('plugin_find_truman_decrypt_1004')
+        default:
+            return ''
+    }
+}
 
 interface EncryptionCardProps {
-    payload: string
+    clueId: string
 }
 export default function EncryptionCard(props: EncryptionCardProps) {
-    const { payload } = props
+    const { clueId } = props
 
-    const { t } = useI18N()
-    const { address } = useContext(FindTrumanContext)
-    const [failed, setFailed] = useState<boolean>(false)
-    const [message, setMessage] = useState<string>('')
+    const { classes } = useStyles()
+    const { address, t } = useContext(FindTrumanContext)
+    const [condition, setCondition] = useState<PuzzleCondition>()
+    const [content, setContent] = useState<string>('')
     const [err, setErr] = useState<{
         code: number
         data: any
     }>()
 
     useEffect(() => {
-        setFailed(false)
-        try {
-            const raw = JSON.parse(Buffer.from(payload, 'base64').toString())
-            if (!!raw.cid && !!raw.data) {
-                fetchSecretKey(raw.cid, address)
-                    .then(async (res) => {
-                        setMessage(raw.data)
-                    })
-                    .catch((error) => {
-                        setErr(error)
-                    })
-            }
-        } catch (error) {
-            setFailed(true)
+        setErr(undefined)
+        setCondition(undefined)
+        if (!!clueId) {
+            fetchClue(clueId, address)
+                .then((res) => {
+                    if (res.decrypted) {
+                        setContent(res.content || '')
+                    } else {
+                        setCondition(res.condition)
+                    }
+                })
+                .catch((error) => {
+                    setErr(error)
+                })
         }
-    }, [payload])
+    }, [clueId])
 
     return (
         <CardContent>
-            {!!message && (
+            {!!content && (
                 <>
                     <Typography variant="body1" color="text.secondary">
                         {t('plugin_find_truman_decrypted_by')}
                     </Typography>
                     <Divider sx={{ margin: '8px 0' }} />
-                    <Typography variant="h6" color="text.primary">
-                        {message}
-                    </Typography>
+                    <CardContent>
+                        <div dangerouslySetInnerHTML={{ __html: content }} className={classes.root} />
```

Comment:
It's dangerous. Better avoid doing this.

Reply:
We have to display rich text here, any good suggestion?